File: run_coqc_complete.js
const { execSync } = require('child_process');
try {
  const result = execSync(
    'cd /workspaces/The-Thiele-Machine/coq && /usr/bin/coqc -Q . Kernel -Q kami_hw KamiHW -Q ../vendor/kami/Kami Kami ThieleMachineComplete.v 2>&1',
    { timeout: 600000, encoding: 'utf-8', maxBuffer: 50 * 1024 * 1024 }
  );
  console.log("OUTPUT:", result);
} catch(e) {
  console.log("STDOUT:", e.stdout);
  console.log("STDERR:", e.stderr);
  console.log("STATUS:", e.status);
  console.log("ERROR:", e.message);
}
const fs = require('fs');
const voPath = '/workspaces/The-Thiele-Machine/coq/ThieleMachineComplete.vo';
if (fs.existsSync(voPath)) {
  const stat = fs.statSync(voPath);
  console.log("VO FILE: exists, size:", stat.size, "mtime:", stat.mtime);
} else {
  console.log("VO FILE: not found");
}

File: run_coqc_complete.py
import subprocess, os, sys

result = subprocess.run(
    ["/usr/bin/coqc", "-Q", ".", "Kernel", "-Q", "kami_hw", "KamiHW",
     "-Q", "../vendor/kami/Kami", "Kami", "ThieleMachineComplete.v"],
    cwd="/workspaces/The-Thiele-Machine/coq",
    capture_output=True,
    text=True,
    timeout=600
)
print("STDOUT:", result.stdout[-3000:] if len(result.stdout) > 3000 else result.stdout)
print("STDERR:", result.stderr[-3000:] if len(result.stderr) > 3000 else result.stderr)
print("RETURN CODE:", result.returncode)
vo_path = "/workspaces/The-Thiele-Machine/coq/ThieleMachineComplete.vo"
if os.path.exists(vo_path):
    st = os.stat(vo_path)
    print(f"VO FILE: exists, size={st.st_size}, mtime={st.st_mtime}")
else:
    print("VO FILE: not found")
